pkg/controller/acmeorders/sync.go

```diff
@@ -473,6 +475,30 @@ func (c *controller) finalizeOrder(ctx context.Context, cl acmecl.Interface, o *
 		return fmt.Errorf("error finalizing order: %v", err)
 	}
 
+	if issuer.GetSpec().ACME != nil && issuer.GetSpec().ACME.PreferredChain != "" {
+		altBundles, err := cl.FetchCertAlternatives(ctx, certURL, true)
+		if err != nil {
+			return fmt.Errorf("error fetching alternate certificates: %w", err)
+		}
+		for _, altBundle := range altBundles {
+			for _, certPEM := range altBundle {
+				cert, err := x509.ParseCertificate(certPEM)
+				if err != nil {
+					return fmt.Errorf("error parsing alternate certificates: %w", err)
+				}
+
+				log.V(logf.DebugLevel).WithValues("Issuer CN", cert.Issuer.CommonName).Info("Found alternative ACME bundle")
+				if cert.Issuer.CommonName == issuer.GetSpec().ACME.PreferredChain {
+					// if the issuer's CN matched the preffered chain it means this bundle is
+					// signed by the requested chain
+					return c.storeCertificateOnStatus(ctx, o, altBundle)
+				}
+			}
+		}
+		// if no match is found we return to the actual cert
+		// it is a *prefered* chain after all
+	}
+
```

Comment:
Does  `FetchCertAlternatives` always include the `certSlice` bundle?
If not, how is a user supposed to `prefer` the `certSlice` chain? By omitting the `PreferredChain` value? That seems unintuitive.

Also, if the user mistypes the `PreferredChain`, then they get back the default chain, without any warning or errors logged.
I'd be inclined to fail the order if the `PreferredChain` is not found, but at least log  and /or  a warning event  so that the user can quickly see that the preferred chain is wrong.

Reply:
>Does FetchCertAlternatives always include the certSlice bundle?
If not, how is a user supposed to prefer the certSlice chain? By omitting the PreferredChain value? That seems unintuitive.

FetchCertAlternatives gives all alternative chains this can be 0 or 100. `PreferredChain` picks the one with the correct root issuer. If not present it will use the "normal" one this allows the ACME server to be free to select which chain it offers in the non alternative reply

Reply:
Ok.